plugins/timeline/index.js

```diff
@@ -0,0 +1,321 @@
+    deferred.resolve();
+  });
+  return deferred.promise;
+};
+
+/**
+ * @param {Object} config The configuration file for the ngHint plugin
+ */
+TimelinePlugin.prototype.teardown = function(config) {
+  var self = this;
+  var deferred = q.defer();
+  // This will be needed later for grabbing data from Sauce Labs.
+  browser.getSession().then(function(session) {
+    self.sessionId = session.getId();
+  });
+
+  // If running with a Selenium Standalone server, get the client logs.
+  if (self.clientLogAvailable) {
+    browser.manage().logs().get('client').then(function(result) {
+      var serverTimeline =TimelinePlugin.parseArrayLog(
+          result, 'Selenium Client', self.testProcessSetTimeoutTimestamp);
+      self.timeline = self.timeline.concat(serverTimeline);
+      deferred.resolve();
+    });
+  } else {
+    deferred.resolve();
+  }
+  return deferred.promise;
+};
+
+/**
+ * @param {Object} config The configuration file for the ngHint plugin
+ */
+TimelinePlugin.prototype.postResults = function(config) {
+  var self = this;
+  var deferred = q.defer();
+  // We can't get Chrome or IE logs from Sauce Labs via the webdriver logs API
+  // because it does not expose them.
+  // TODO - if the feature request at
+  // https://support.saucelabs.com/entries/60070884-Enable-grabbing-server-logs-from-the-wire-protocol
+  // gets implemented, remove this hack.
+  if (config.sauceUser && config.sauceKey) {
+    // WARNING, HACK: we have a timeout to deal with the fact that there's a
+    // delay before Sauce Labs updates logs.
+    setTimeout(function() {
+      var sauceServer = new SauceLabs({
+        username: config.sauceUser,
+        password: config.sauceKey
+      });
+
+      sauceServer.showJob(self.sessionId, function(err, job) {
+        var sauceLog = '';
+        if (!job.log_url) {
+          console.log('WARNING - no Sauce Labs log url found');
+          deferred.resolve();
+          return;
+        }
+        https.get(job.log_url, function(res) {
+          res.on('data', function(data) {
+            sauceLog += data;
+          });
+
+          res.on('end', function() {
+            var sauceTimeline =
+                TimelinePlugin.parseTextLog(
+                    sauceLog,
+                    'SauceLabs Server',
+                    self.testProcessSetTimeoutTimestamp);
+            self.timeline = self.timeline.concat(sauceTimeline);
+            self.outputResults(deferred.resolve);
+          });
+
+        }).on('error', function(e) {
+          console.error(e);
+        });
+      });
+    }, SAUCE_LOGS_WAIT);
+  } else {
+    self.outputResults(deferred.resolve);
+  }
+  return deferred.promise;
+};
+
+
+// Export
+
+var timelinePlugin = new TimelinePlugin();
+
+exports.setup = timelinePlugin.setup.bind(timelinePlugin);
+exports.teardown = timelinePlugin.teardown.bind(timelinePlugin);
+exports.postResults = timelinePlugin.postResults.bind(timelinePlugin);
+exports.TimelinePlugin = TimelinePlugin;
```

Comment:
I put the e2e tests in the timeline folder and added to the specs script. It doesn't really test much except that it doesn't blow up and fail, but the unit tests test the actual parsing.